src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -325,6 +325,11 @@ internal void RegisterBuiltInSerializers()
             Register(typeof(IList<>));
             Register(typeof(IDictionary<,>));
             Register(typeof(IEnumerable<>));
+            Register(typeof(ICollection<>));
+            Register(typeof(ISet<>));
+            Register(typeof(IReadOnlyDictionary<,>));
+            Register(typeof(IReadOnlyCollection<>));
+            Register(typeof(IReadOnlyList<>));
```

Comment:
shouldn't be, since this won't change how anything is actually serialized, only what's permissible during deserialization